### PR TITLE
[SPARK-32682][INFRA] Use workflow_dispatch to enable manual test triggers

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This PR proposes to add a `workflow_dispatch` entry in the GitHub Action script (`build_and_test.yml`). This update can enable developers to run the Spark tests for a specific branch on their own local repository, so I think it might help to check if al the tests can pass before opening a new PR.

Screen Shot 2020-08-21 at 16 28 41

### Why are the changes needed?

To reduce the pressure of GitHub Actions on the Spark repository.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manually checked.

### CODE REVIEW

**Comment by maropu** [comment]
@HyukjinKwon @dongjoon-hyun How about this update? Recently, I started checking whether all the related tests (by using a modified GitHub Actions script) can pass in my local repository before opening a PR. (Note: I'll file a jira later if necessary)

**Review by dongjoon-hyun — COMMENTED** [review]
So, is it useful when we test before making a PR?

**Comment by dongjoon-hyun** [comment]
Please file a Jira. I also want this feature!!! 😀

**Comment by HyukjinKwon** [comment]
Yeah .. I want this feature too

**Comment by HyukjinKwon** [comment]
This can also leverage developer' own GitHub Actions resources and save the ASF repos'. Why not!

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -7,6 +7,11 @@ on:
   pull_request:
     branches:
     - master
+  workflow_dispatch:
+    inputs:
+      target:
+        description: 'Target branch to test'
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:13]: no big deal but I would call it `Target branch to run`.

**Comment by HyukjinKwon** [comment]
Looks good otherwise if it works correctly.

**Comment by maropu** [comment]
> So, is it useful when we test before making a PR?

Thanks, @dongjoon-hyun @HyukjinKwon ! Yea, I think it is still useful; sometimes I see some developers debugging on PRs after opening PRs, but I think this can burden our limited resources of  Github Action...

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (maropu) [reply .github/workflows/build_and_test.yml:13]: okay, I'll update.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
@maropu, WDYT about sending a short email to the dev mailing list about this new dev feature? I think it's really useful.

**Comment by maropu** [comment]
Yea, okay, @HyukjinKwon ! I'll do it later.

**Pull request closed without merging**